scripts/pipeline/orthology_benchmark.py

```diff
@@ -157,12 +160,49 @@ def get_core_names(species_names: List[str], host: str, port: int, user: str) ->
     return core_names
 
 
-def prep_input_for_orth_tools():
-    """Docstring"""
+def prep_input_for_orth_tools(source_dir: str, target_dir_ofinder: str) -> None:
+    """Prepares input files for selected orthology inference tools.
 
+    For OrthoFinder: Creates symlinks to input fasta files in `target_dir_ofinder`.
+    # For OMA: ...
 
-def run_orthology_tools():
-    """Docstring"""
+    Args:
+        source_dir: Path to the directory containing fasta files.
+        target_dir_ofinder: Path to the directory where symlinks to fasta files will be created
+            for OrthoFinder to use.
+
+    Raises:
+        FileExistsError: If directory `target_dir_ofinder` already exists.
+        FileNotFoundError: If directory `source_dir` does not exist or does not contain any fasta files.
+        subprocess.CalledProcessError: If creating symlinks fails for some other reason.
+    """
+    if not os.path.isdir(source_dir):
+        raise FileNotFoundError("Directory containing fasta files not found.")
+    if len(fnmatch.filter(os.listdir(source_dir), '*.fasta')) == 0:
+        raise FileNotFoundError("No fasta files found.")
+
+    # OrthoFinder
+    os.mkdir(target_dir_ofinder)
```

Comment:
The `symlink_fasta.py` script does this for you:
```suggestion
```

Reply:
`symlink_pasta.py` does `Path(symlink_dir).mkdir(parents=True, exist_ok=True)`. So, let's say that there was already a dir `dummy` with some other unrelated analyses and a user mistakenly sets it for `target_dir_ofinder`. 
- `os.mkdir(target_dir_ofinder)` would throw an error because the directory already exists. The script would stop and no symlinks would be created.
- If that line would be removed, `symlink_fasta.py` would create symlinks in `dummy` and the pipeline would continue running, potentially messing up its already existing content.

Reply:
I am not sure that is a real problem. It's a technical possibility, in which there are hundreds we cannot possibly think of, but rather unlikely to happen. This is not far off what occurs in almost every single production pipeline we run, with `rm_mkdir` - in which the directories are silently overridden. In this case, there is the potential issue that orthofinder or other-orthotool is run on files we would not want it to run on, but these tools do not alter fasta files and worst case scenario is producing data we do not want. If this particular script was moved into a workflow manager such as NextFlow or SnakeMake the equivalent type of method would be used for creating the new directory too.

Reply:
A side note on this, if you see potential pitfalls in a compara script you are using in your code, you are encouraged to tweak it to fit your purposes, it makes it more generic, and it could be more useful to other purposes too - for example, passing in a flag to `symlink_fasta.py` to force it to die if the directory already exists.

Reply:
What's the problem with ensuring that OrthoFinder is run on the dataset the user wants?

Reply:
That isn't a problem, the user passes in the directory via the command line - so it should be the dataset the user wants to run it on. My suggestions are based on redundancy and my comments in response to your explanation are to further support my suggestion, but also to include that if you see it as an issue, perhaps it would be nice for you to alter the existing code in `symlink_fasta.py` to fit your purpose.

Reply:
Yes, you're right - the user passes in the directory, thus it's prone to human error. Better safe than sorry.

BTW, it'd be great to extend the functionality of `symlink_fasta.py` (could also add the option for `.fa` files). At the moment it still looks like work in progress (no usage example, no pytest) so once that's done, I can extend it with the bits that'd be useful for `orthology_benchmark.py`. pytest would be helpful for future developments to ensure that the new code doesn't mess up the existing functionality which is desired to remain or be improved.

As of `os.mkdir(target_dir)`, I also have similar safety measure elsewhere in this script, e.g. https://github.com/Ensembl/ensembl-compara/pull/364/files#r750455647

Anyways, I'll go through the whole pipeline when moving it to NextFlow, some Python stuff will be removed so that will be a good time for a quick iteration of further polishing.